### PR TITLE
Homepage cleanup

### LINKED ISSUE #43494: Quick .com/search optimizations
We want to drastically reduce the noise below the search bar so that devs can start searching and *using* Sourcegraph right away. We need to figure out the best way to help them search (that's what we're working with design on in https://github.com/sourcegraph/sourcegraph/issues/43255), but we can get rid of *lots* of stuff right away. 

For unauth'd, lets: 
- Remove video
- Remove tour
- Remove the big banner that directs to the About site
- Replace the big Cloud CTA banner with just a link that says, "Try Sourcegraph on your code"
- Remove the OSS logos
- Remove the customer logos

For both unauth'd and auth'd, lets: 
- Remove the Discord logo in the footer (just the existing copy is fine)
- Replace the text in the CTA in the unauth'd nav (now "Get Free Trial") with "Search your code" and add that same CTA to the auth'd page. The thinking is that a dev would see "Get Free Trial" and think, "What does that mean? I'm already trying Sourcegraph."

### PR DESCRIPTION
**Pull request by st0nebraker**
Closes #43494

Removes
- video
- tour
- big banner that directs to the About site
- OSS logos
- customer logos
- Discord logo in footer
- Replaces the text in the CTA "Get Free Trial" with "Search your code"
- Trades out Cloud CTA banner for simplified version ([Figma](https://www.figma.com/file/qdUtoveMtn2wI8eaysE1De/UX%3A-Cleaning-up-noise-below-the-search-bar-on-dotcom?node-id=51%3A4427))

## Test plan
- Test locally on dotcom

## App preview:

- [Web](https://sg-web-becca-homepage-cleanup.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (+0.00 kb) | -0.40% (-58.73 kb) 🔽 | -0.50% (-58.73 kb) 🔽 | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/becca/homepage-cleanup/compare-report.html;tab=live_object) for a full comparison between the commits fa92fa7 and d30f25b or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: client/web/src/search/home/LoggedOutHomepage.module.scss [hunk 1]
```diff
@@ -18,49 +18,10 @@
     --home-marketing-drop: 0 4px 16px -6px rgba(46, 34, 119, 0.1);
     --search-input-shadow: 0 1px 2px rgba(0, 0, 0, 0.04);
 
-    &__community-search-context-list {
```
Comment (limitedmage) [inline client/web/src/search/home/LoggedOutHomepage.module.scss:21]: If we're removing these, do we want to consider removing community search contexts altogether?

File: client/web/src/search/home/SearchPageFooter.tsx [hunk 2]
```diff
@@ -105,7 +105,6 @@ export const SearchPageFooter: React.FunctionComponent<
                         className={styles.discordWrapper}
                         onClick={logDiscordClicked}
                     >
-                        <img src={`${assetsRoot}/img/discord-footer-logo.svg`} alt="" className={styles.discordImage} />
                         <div className={styles.discordText}>
                             <H2 className={styles.linkSectionHeading}>Join our Discord</H2>
                             <div>If you need help or want to share something with the community, join us!</div>
```
Comment (limitedmage) [inline client/web/src/search/home/SearchPageFooter.tsx:110]: It looks a bit odd that the whole thing is clickable and gets underlined. Maybe only the `h2` should be the link?

**Review by limitedmage — APPROVED** [review]

File: client/web/src/search/home/SearchPageFooter.tsx (hunk 2 above)
Comment (st0nebraker) [reply client/web/src/search/home/SearchPageFooter.tsx]: Totally agree, fixed that in [8a1b6ff](https://github.com/sourcegraph/sourcegraph/pull/43507/commits/8a1b6ff44295f281eec14766815ee50f9a74be08) 😄

File: client/web/src/search/home/LoggedOutHomepage.module.scss (hunk 1 above)
Comment (taylorsperry) [reply client/web/src/search/home/LoggedOutHomepage.module.scss:21]: I think the community search contexts could still be useful in a different format. Discussion [here](https://www.figma.com/file/qdUtoveMtn2wI8eaysE1De?node-id=51:4940#292116998).

**Comment by st0nebraker** [comment]
Ready for another review @limitedmage  - note CTA banner traded out for a simplified version ([Figma](https://www.figma.com/file/qdUtoveMtn2wI8eaysE1De/UX%3A-Cleaning-up-noise-below-the-search-bar-on-dotcom?node-id=51%3A4427))

File: client/web/src/search/home/LoggedOutHomepage.module.scss (hunk 1 above)
Comment (st0nebraker) [reply client/web/src/search/home/LoggedOutHomepage.module.scss:21]: [Context](https://sourcegraph.slack.com/archives/C046X8AG5NY/p1666904313684779?thread_ts=1666903621.302829&cid=C046X8AG5NY)

**Review by sqs — COMMENTED** [review]
Awesome! I don't see the example search queries anymore, though. I think those would be good to keep.

image

I mean the things that look like:
image

(If adding those in is separate, just disregard my comment.)

**Comment by st0nebraker** [comment]
> * Web

@sqs It has to be run locally to view dotcoms version. This is what I'm getting:

image

The unauth'd dotcoms code basics & example queries will come in #43570 & #43572. Let me know if I'm missing something?

**Comment by taylorsperry** [comment]
FYI, the "Code search basics" in this PR are temporary. We're fleshing them out to be more like this. I think having the design for it finalized tomorrow is reasonable. 
Screen Shot 2022-10-27 at 4 34 32 PM

**Comment by sqs** [comment]
@st0nebraker Ahh, makes sense, I forgot the preview URL in the PR description wouldn't have dotcom mode set to on.

**Review by sqs — APPROVED** [review]
(No need to wait for my approval, but I'm approving since you requested. Looks awesome, thank you!)

**Review by limitedmage — APPROVED** [review]
Looks great!

**Comment by paulo-digital** [comment]
Nice work!

**Pull request merged**